Bugzilla/Bug.pm

```diff
@@ -1784,7 +1786,7 @@ sub _check_bug_status {
     @valid_statuses = @{$invocant->statuses_available};
     $product        = $invocant->product_obj;
     $old_status     = $invocant->status;
-    my $comments = $invocant->{added_comments} || [];
+    my $comments = $invocant->{pending_comments} || [];
```

Comment:
If the `_check*` methods are called after the first `$bug->update()` then they will incorrectly report that no comments have been added as the comments would have been moved from `pending_comments` to `added_comments`.